### PR TITLE
refactor(clickhouse_handler): use new InputFormat.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

I will remove unused code after this pr is merged

Fixes #issue

part of https://github.com/datafuselabs/databend/issues/7732

### CODE REVIEW

File: src/meta/types/src/user_stage.rs [hunk 1]
```diff
@@ -135,17 +135,16 @@ impl FromStr for StageFileFormatType {
...
             "AVRO" => Ok(StageFileFormatType::Avro),
             "ORC" => Ok(StageFileFormatType::Orc),
             "PARQUET" => Ok(StageFileFormatType::Parquet),
             "XML" => Ok(StageFileFormatType::Xml),
-            _ => Err(
-                "Unknown file format type, must one of { CSV | JSON | AVRO | ORC | PARQUET | XML }"
-                    .to_string(),
-            ),
+            _ => Err(format!(
+                "Unknown file format type '{s}', must one of ( CSV | TSV | JSON | AVRO | ORC | PARQUET | XML)"
```
Comment (Xuanwo) [inline src/meta/types/src/user_stage.rs:146]: I don't know we support so many formats...

Not included in docs: https://databend.rs/doc/reference/sql/dml/dml-copy-into-table

File: tests/suites/0_stateless/14_clickhouse_http_handler/14_0007_http_clickhouse_input_format_diagnostic.sh [hunk 2]
```diff
@@ -3,58 +3,60 @@
...
-echo -e '\n'
-curl -s -u 'root:' -XPOST "http://localhost:${QUERY_CLICKHOUSE_HTTP_HANDLER_PORT}/?enable_planner_v2=1" --data-binary @/tmp/databend_test_tsv_error2.txt
-echo -e '\n'
-curl -s -u 'root:' -XPOST "http://localhost:${QUERY_CLICKHOUSE_HTTP_HANDLER_PORT}/?enable_planner_v2=1" --data-binary @/tmp/databend_test_tsv_error3.txt
-
+# cleanup
 curl -s -u 'root:' -XPOST "http://localhost:${QUERY_CLICKHOUSE_HTTP_HANDLER_PORT}/?enable_planner_v2=1" -d "drop table a"
-
-rm /tmp/databend_test*.txt
+# rm /tmp/databend_test*.txt
```
Comment (Xuanwo) [inline tests/suites/0_stateless/14_clickhouse_http_handler/14_0007_http_clickhouse_input_format_diagnostic.sh:62]: remove this line?

File: src/meta/types/src/user_stage.rs (hunk 1 above)
Comment (youngsofun) [reply src/meta/types/src/user_stage.rs:146]: we do not.. 
@BohuTANG added this.

we will support these in the further

File: tests/suites/0_stateless/14_clickhouse_http_handler/14_0007_http_clickhouse_input_format_diagnostic.sh (hunk 2 above)
Comment (youngsofun) [reply tests/suites/0_stateless/14_clickhouse_http_handler/14_0007_http_clickhouse_input_format_diagnostic.sh:62]: no, I change it when debuging, forgot to change back

**Comment by youngsofun** [comment]
@Xuanwo forgot to change to Draft when fixing failed tests. rebased just to merge a small fix commit to earlier commit, no big change.

**Comment by youngsofun** [comment]
flaky test test_alter_recluster_interpreter
https://github.com/datafuselabs/databend/actions/runs/3112438121/jobs/5045863994

cc @zhyass

**Review by Xuanwo — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**